### DISCUSSION TITLE
100% self-hosted instant messaging solution compatible with Wiki.js

### DISCUSSION THREAD

**Discussion by Alexis-G2A** [requarks/wiki#7300] [Help / Questions]
Hi,

I would like to propose a wiki solution for the company i'm working for but it need to be 100% self-hosted, that why i will propose Wiki.js.

But i would also like to propose an instant messaging module with, but it need to be 100% self-hosted also, do you have a tool or a module like that ? 

Thank in advance for your help

**Comment by johndickinson** [comment]
What do you mean by instant messaging solution compatible with Wiki.js?
What do you mean by instant messaging? jabber, slack/mattermost or something else?

**Reply by Alexis-G2A** [reply]
Hi sir,

Thank you for your answer, and yes something like slack, it's would be just for chatting, no call or visio.
Because i am new to this type of tools, and need to be 100% self hosted, there is a lot of information to found and learn.

And i guess that is maybe a good thing to have your wiki and you chat on the same tools.

**Reply by johndickinson** [reply]
well if you want something like slack then I don't think you will find better than Mattermost. You can self host it for free if you don't want any of the paid for extras. I don't know if you will find all the integration you want but I think both have various plugins. Also, Mattermost does have web hooks. IMHO I think both Wiki.js and Mattermost are the best at what they do, I just hope you can find the integrations you need.

**Reply by Alexis-G2A** [reply]
Thank you very much for all those informations and your opinion, please have a really good day !

**Comment by Kathrininfanta** [comment]
Hi, in my opinion, there isn’t a built-in self-hosted instant messaging module for [Wiki.js](http://wiki.js/). You can workaround by integrating any self-hosted solution along with [wiki.js](http://wiki.js/). For example, platform such as Rocket.Chat, Mattermost, MirrorFly can fit your need. I would suggest you to embed chat widgets into your wiki pages via iframe or custom HTML or by using SSO for a seamless user experience. Hope it helps.

I found the source from this blog: [Best Instant messaging platform](https://www.apphitect.ae/blog/instant-messaging-platforms/)

**Comment by alexsam9866** [comment]
Hi, if you’re aiming for a 100% self-hosted setup, you’ve got a few good options for the chat side. Solutions such as MirrorFly and Apphitect are ready-made instant messaging apps. You can run everything on your own infrastructure and integrate it with your internal tools or even your wiki. Sendbird is popular, though it’s more cloud-focused.

This comparison list of the [Best Free Real-time Chat APIs](https://www.trustfirms.com/best-free-chat-apis-sdks) would help you evaluate options. Thank me later.
